pyrealm/splash/splash.py

```diff
@@ -141,18 +144,20 @@ def estimate_initial_soil_moisture(
 
         if wn_init is not None:
             # Check the shape is the same as the shape of a slice along axis 0
-            if wn_init.shape != self.tc[0].shape:
+            if wn_init.shape != self.shape[1:]:
```

Comment:
Is there a reason for making these changes to the shape checking (and also `np.arange` to `range`). As far as I can see they behave identically, so I'm not sure why they have been altered.

Reply:
I feel it's better to use `self.shape` and make the shape checking independent of a specific variable name. The replacements of `np.arange` by `range` happen to occur in the same lines. I'm not sure why `np.arange` is preferred over `range`.